Dockerfile

```diff
@@ -7,6 +7,9 @@ ARG BASE_IMAGE=debian:9.5-slim
 FROM golang:1.12.6 as builder
 
 RUN apt-get update && apt-get install -y \
```

Comment:
Should we create a completely separate image for this rather than bundling everything together? One you could just `docker run`?

Reply:
The builder stage is supposed to have all dependencies which are required to build/test argocd.
We used to maintain a separate image and then switched to multi-stage build hoping to reduce maintenance overhead.

Reply:
Unfortunately, multi-stage builds don't work so well: we've tried using `--cache-from` but it appears to be no reliable, so everyone has to rebuild builder locally. I never could make cache work in CI. I would move the whole builder into separate dockerfile in a separate PR.